gpt4all-chat/modellist.cpp

```diff
@@ -1457,7 +1457,7 @@ void ModelList::parseModelsJsonFile(const QByteArray &jsonData, bool save)
     {
         const QString modelName = "ChatGPT-3.5 Turbo";
         const QString id = modelName;
-        const QString modelFilename = "chatgpt-gpt-3.5-turbo.txt";
```

Comment:
at the least we need to keep these files associated with the OpenAI so it is clear what they are...

Reply:
the problem with this is : currently, the filename matches the model name for the api. It is indeed more clear to add "chatgpt" but it doesn't match the model name anymore. We would have to create a new parameter for the model name to fix this

Reply:
I need to have the api model's name stored somewhere. The "modelName" parameter is the display name, and therefore cannot use it. I didn't want to add a new parameter specifically for apis, so I used the filename. I know it's not an ideal solution, but I don't know how to handle it properly. Tell me if you have any idea on fixing that

Reply:
I understand. I think you should add the model name to the filename at installation time as we discussed above. That means the format of the file has to be changed to include not just the api key but also the model name and potentially the url later and other additional stuff.

I think we should go with either an INI file or a JSON format. I'll leave it to the implementer to decide. I think we should also go with 'gpt4all-' as a prefix and '.rmodel' as a file extension.

I hope that is clear enough guidance to resolve the remaining issues and get this merged in. Let me know.